tests/testsuite.src/run_misc.at

```diff
@@ -15251,3 +15251,324 @@ AT_CHECK([$COMPILE prog.cob])
+TestRecurse,,,,prog.cob:28,PROGRAM,21
+TestRecurse,MAIN SECTION,,,prog.cob:29,SECTION,21
+TestRecurse,MAIN SECTION,MAIN PARAGRAPH,,prog.cob:29,PARAGRAPH,21
+TestRecurse,MAIN SECTION,MAIN PARAGRAPH,TestRecurse,prog.cob:37,CALL,15
+PROG,,,,prog.cob:10,PROGRAM,1
+PROG,MAIN SECTION,,,prog.cob:11,SECTION,1
+PROG,MAIN SECTION,MAIN PARAGRAPH,,prog.cob:11,PARAGRAPH,1
+PROG,MAIN SECTION,MAIN PARAGRAPH,TestRecurse,prog.cob:13,CALL,6
+], [])
+
+
+AT_CLEANUP
+
+
+AT_SETUP([INDEXED BY in LINKAGE])
+
+AT_KEYWORDS([runmisc OCCURS SEARCH])
+
+AT_DATA([callee.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      callee.
+       DATA             DIVISION.
+       LINKAGE          SECTION.
+           01 LK-STR.
+              03 LK-CHR PIC X OCCURS 6 INDEXED BY LK-IDX.
+           01 LK-OFFSET BINARY-LONG UNSIGNED.
+       PROCEDURE        DIVISION USING LK-STR LK-OFFSET.
+           DISPLAY "LK-OFFSET: " LK-OFFSET.
+           SET LK-IDX TO LK-OFFSET.
+           SEARCH LK-CHR
+               AT END
+                   IF LK-OFFSET > 20
+                      GOBACK
+                   END-IF
+                   ADD 1 TO LK-OFFSET
+               WHEN LK-CHR (LK-IDX) = "."
+                   ADD 1 TO LK-OFFSET
+           END-SEARCH.
+           EXIT PROGRAM.
+])
+
+AT_DATA([caller.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      caller.
+       DATA             DIVISION.
+       WORKING-STORAGE SECTION.
+           01 STR PIC X(6) VALUE "..-..-".
+           01 OFFSET BINARY-LONG UNSIGNED.
+       PROCEDURE        DIVISION.
+           MOVE 1 TO OFFSET.
+           PERFORM UNTIL OFFSET > LENGTH OF STR
+               CALL "callee" USING STR OFFSET
+           END-PERFORM.
+])
+
+AT_CHECK([$COMPILE_MODULE callee.cob], [0], [], [])
+AT_CHECK([$COMPILE -o prog caller.cob], [0], [], [])
+AT_CHECK([$COBCRUN_DIRECT ./prog], [0], [LK-OFFSET: 0000000001
+LK-OFFSET: 0000000002
+LK-OFFSET: 0000000003
+LK-OFFSET: 0000000004
+LK-OFFSET: 0000000005
+LK-OFFSET: 0000000006
+], [])
+
+AT_CLEANUP
+
+
+AT_SETUP([profiling RECURSIVE CALL])
+
+AT_DATA([prog.cob], [
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. prog.
+       DATA DIVISION.
+       WORKING-STORAGE SECTION.
+       01 WS-COUNTER PIC 9.
+       PROCEDURE DIVISION.
+           MOVE 0 TO WS-COUNTER.
+           CALL "recurse" USING WS-COUNTER.
+           STOP RUN.
+       END PROGRAM prog.
+
+       IDENTIFICATION DIVISION.
+       PROGRAM-ID. recurse IS RECURSIVE.
+       DATA DIVISION.
+       LINKAGE SECTION.
+       01 LK-COUNTER PIC 9.
+       PROCEDURE DIVISION USING LK-COUNTER.
+           DISPLAY "COUNTER:" LK-COUNTER.
+           EVALUATE LK-COUNTER
+              WHEN 5
+                EXIT PROGRAM
+              WHEN OTHER
+                ADD 1 TO LK-COUNTER
+                CALL "recurse" USING LK-COUNTER
+           END-EVALUATE.
+       END PROGRAM recurse.
+])
+
+AT_CHECK([$COMPILE --free -fprof prog.cob])
```

Comment:
same as above, no need for free, less output if we only display 1 + 5  (maybe even WITH NO ADVANCING and no text...)